selenium_testcase/testcases/find.py

```diff
@@ -42,11 +50,16 @@ def find_element(self, search_list, *args, **kwargs):
 
             # try each of the patterns in order
             try:
-                element = self.browser.find_element(method, value)
+                elements = self.browser.find_elements(method, value)
             except NoSuchElementException:
                 pass
             else:
-                return element
+                # continue if we don't have enough elements returned
+                if len(elements) < index + 1:
+                    continue
+                # return element if it is visible
+                if elements[index].is_displayed():
```

Comment:
This condition doesn't make sense to me. We're returning a list of items but only ensuring visibility on one of them? That seems to make the index somewhat meaningless if there is a mixture of visible and not visible elements on the page.

It seems that we should be checking visibility on every item, (adding only visible elements to the list and then indexing the Nth visibile item) or else not checking visibility at all.